Starts logging

cptwonton opened the pull request:
adds documentation for logging functionality in STARTS

includes info about artifact storage from paper

adds link in README.md to starts logging docs

cptwonton committed:
Create STARTS-LOGGING.md

adds logging documentation to STARTS, incorporating the paper's details on artifact storage

august782 commented on README.md:
```diff
@@ -66,6 +66,8 @@ checksums of files in the latest version, while the command in (4)
 behavior. For example, to update the checksums while checking the
 diff, run `mvn starts:diff -DupdateDiffChecksums=true`.
 
+[Logging and Artifact storage Docs](./STARTS-LOGGING.md)
```
Capitalize storage

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+For any piece of starts that you'd like to add logging to, begin by adding two import statements at the top:
+-  ``import edu.illinois.starts.util.Logger;``
+- ``import java.util.logging.Level;``
+_note: you may need to make minor changes to the positioning of these two import statements to ensure checkstyle does not break. place ``edu.illinois.starts.util.Logger`` directly underneath the other imports with the same package name, and do the same with ``java.util.logging.Level``. additionally, ensure you have a newline separating different package names_
```
Capitalize place, capitalize additionally.

Is there a way to get this note to appear underneath the previous point (import java.util.logging.Level;), indented along with it, but without the bullet point? If possible, I think it would look nicer.

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+- FINEST (lowest value)
+
+To set the logging level of your log, use the ``setLoggingLevel(Level level)`` method.
+i.e.
```
i.e. -> i.e.,

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+``logger.setLoggingLevel(Level.CONFIG);``
+
+To check the logging level, use the ``getLoggingLevel()`` method, which will return an object of type [Level](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Level.html).
+i.e.
```
i.e. -> i.e.,

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+###### ``public void log(Level lev, String msg)``
+should be used when you only want to have a custom message
+
+i.e. ``logger.log(Level.SEVERE, "houston we have a problem");``
```
i.e. -> i.e.,

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+## Where will messages be output? 
+Standard Output (System.out)
+
+## Artifact Storage
```
Uncapitalize Storage (for consistency of the other sections).

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+
+``mvn starts:starts -DStartsLogging=<Level>``
+
+i.e. ``mvn starts:starts -DStartsLogging=FINEST``
```
i.e. -> i.e.,

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+- dependency file (.starts/deps.zlc)
+
+Level.FINER will store:
+- dependency file (.starts/deps.zlc)
```
Shouldn't FINER include all the things in INFO, including checksum (actually, isn't checksum a part of dependency file)?

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+- list of impacted tests
+
+Level.FINEST will store:
+- list of all tests
```
Similarly, shouldn't FINEST also contain of of FINER (currently missing dependency file/checksum)?

august782 commented on STARTS-LOGGING.md:
```diff
@@ -0,0 +1,82 @@
+
+To set the log level at runtime, call starts like this: 
+
+``mvn starts:starts -DStartsLogging=<Level>``
```
I believe the option actually has it lower case, i.e., startsLogging. Please double check.

cptwonton committed:
adds fixes suggested by August

cptwonton commented:
@august782 thanks for the feedback. I've rebased to include the two merges that Owolabi recently made to testingresearchillinois/starts, and added the fixes you've suggeted

owolabileg commented:
@cptwonton, please rebase [master](https://github.com/TestingResearchIllinois/starts/tree/master), into this branch, and we can see if it is ready to merge.

cptwonton commented:
@owolabileg this is done

cptwonton commented:
@owolabileg rebased and squashed and ready for merging into master

benjamin-shen commented:
Closes #4

benjamin-shen reviewed (commented):
Has there been any discussion on whether or not this documentation should live in another place, like a `docs/` folder (or even the Github wiki)?

@owolabileg other than my comments, this pr LGTM.

owolabileg committed:
capitalization

Co-authored-by: Benjamin Shen <34158284+benjamin-shen@users.noreply.github.com>

owolabileg committed:
stylize URL

Co-authored-by: Benjamin Shen <34158284+benjamin-shen@users.noreply.github.com>
